starknet_py/tests/e2e/account/account_test.py

```diff
@@ -328,3 +331,88 @@ async def test_deploy_account_uses_custom_calldata(
     tx = await client.get_transaction(deploy_result.hash)
     assert isinstance(tx, DeployAccountTransaction)
     assert tx.constructor_calldata == calldata
+
+
+@pytest.mark.asyncio
+async def test_sign_invoke_tx_for_fee_estimation(account, map_contract):
+    call = map_contract.functions["put"].prepare(key=40, value=50)
+    transaction = await account.sign_invoke_transaction(calls=call, max_fee=MAX_FEE)
+
+    signed_transaction = await account.sign_for_fee_estimate(transaction)
+
+    estimation = await account.client.estimate_fee(signed_transaction)
+    assert estimation.overall_fee > 0
+
+    # Verify that the transaction signed for fee estimation cannot be sent
+    with pytest.raises(ClientError):
+        await account.client.send_transaction(signed_transaction)
+
+    # Verify that original transaction can be sent
+    result = await account.client.send_transaction(transaction)
+    await account.client.wait_for_tx(result.transaction_hash)
+
+
+@pytest.mark.asyncio
+async def test_sign_declare_tx_for_fee_estimation(account, map_compiled_contract):
+    transaction = await account.sign_declare_transaction(
+        compiled_contract=map_compiled_contract, max_fee=MAX_FEE
+    )
+
+    signed_transaction = await account.sign_for_fee_estimate(transaction)
+
+    estimation = await account.client.estimate_fee(signed_transaction)
```

Comment:
That flow is the default for estimating fees now. It means a user has to sign the transaction properly and then resign it for estimation. What if someone only wants to estamete_fee? Isn't it too complex? Shouldn't we add an option to create a transaction from scratch here? Any ideas?

Reply:
eh. Well I suppose if someone estimates the fee of a transaction, they probably do want to send it later.

Reply:
An optional flag would keep the flow the same, but the method would have to be used twice when wanting to send the tx, so no benefit.

Reply:
I think there are little to now use-cases for creating transactions only for purpose of fee estimation.